test/sparsity/test_sparse_api.py

```diff
@@ -37,7 +38,7 @@ def test_sparse(self):
         apply_fake_sparsity(model)
         dense_result = model(input)
 
-        apply_sparse_semi_structured(model)
+        model = sparsify(model, to_sparse_semi_structured)
```

Comment:
is sparsify an in place op? This came up recently since `quantize` is in place and here it looks like the api used to be in place but now its not

Reply:
Yeah, I think there's some [discussion](https://github.com/pytorch/ao/pull/467#discussion_r1664508006) on whether we can use `quantize_` or `quantize` for the in-place op, I'm not sure if we came to a conclusion. cc @jerryzh168 do you have a preference for what to use here?

Reply:
we can use the inplace version for now I think

Reply:
@jcaip I think we can change this to `sparsify_` to be consistent with quantization: https://github.com/pytorch/ao/pull/467